src/Components/NotesPage/NotesPage.jsx

```diff
@@ -1,80 +1,126 @@
 		const editorState = EditorState.createWithContent(contentState);
 		setEditorState(editorState);
 	};
 
+	const { FilterState } = useFilter();
+
 	useEffect(() => {
-		setTextField({ ...textField, body: draftToHtml(convertToRaw(editorState.getCurrentContent())) });
+		setFormData({ ...formData, body: draftToHtml(convertToRaw(editorState.getCurrentContent())) });
 	}, [editorState]);
 
 	return (
-		<div className='app-content'>
-			<button onClick={() => toggleModal()} className='btn btn-fab'>
-				<CreateNewIcon />
-			</button>
-			<Modal showModal={modal}>
-				<AddNewNoteCard
-					onEditorStateChange={onEditorStateChange}
-					titleValue={textField.title}
-					submitNote={submitNote}
-					editorState={editorState}
-					titleChangeHandler={titleChangeHandler}
-				/>
-			</Modal>
-
-			{notes.length > 0 ? (
-				notes.map((note) => {
-					return <NotesCard note={note} key={note._id} />;
-				})
-			) : (
-				<h1>Add Notes here</h1>
-			)}
-		</div>
+		<>
+			<div className='notes-body'>
+				{notes.length > 0 && <FilterBar />}
+				<div className='notes-content'>
+					<button onClick={() => toggleModal()} className='btn btn-fab flex-column justify-content-center align-items-center'>
+						<CreateNewIcon />
+					</button>
+					<Modal showModal={modal}>
+						<AddNewNoteCard
+							formData={formData}
+							editorState={editorState}
+							onEditorStateChange={onEditorStateChange}
+							submitNote={submitNote}
+							titleChangeHandler={titleChangeHandler}
+							tagChangeHandler={tagChangeHandler}
+							priorityChangeHandler={priorityChangeHandler}
+							colorChangeHandler={colorChangeHandler}
+						/>
+					</Modal>
+
+					{notes.length > 0 ? (
+						getFilterData(FilterState).length > 0 ? (
+							getFilterData(FilterState).map((note) => {
+								return <NotesCard note={note} key={note._id} isHomePage={true} />;
+							})
+						) : (
+							<h1>No notes with applied filter</h1>
+						)
+					) : (
+						<h1>Add Notes here</h1>
+					)}
+				</div>
+			</div>
+		</>
 	);
 };
 
 export { NotesPage };
+
+const getFilterData = (filterState) => {
+	const { tag, priority, search, dataToShow } = filterState;
+	const dataByTag = getDataByTag(tag, dataToShow);
+	const dataByPriority = getDataByPriority(priority, dataByTag);
+	const dataBySearch = getDataBySearch(search, dataByPriority);
+	return dataBySearch;
+};
+
+const getDataByTag = (selectedTag, data) => (selectedTag === 'None' || selectedTag === '' ? data : data.filter((note) => note.tag === selectedTag));
+
+const getDataByPriority = (selectedPriority, data) =>
+	selectedPriority !== '' && selectedPriority !== 'All' ? data.filter((note) => note.priority == selectedPriority) : data;
+
+const getDataBySearch = (searchedWord, data) => {
+	return searchedWord !== null && searchedWord !== '' ? data.filter((note) => note.title.toLowerCase().includes(searchedWord.toLowerCase())) : data;
+};
```

Comment:
You can also add an extra trim() to ignore trailing spaces in the search keyword.